Packages/Tracking/Core/Runtime/Scripts/LeapServiceProvider.cs

```diff
@@ -1229,13 +1229,12 @@ private void OnDrawGizmos()
 
         private void DetectConnectedDevice(Transform targetTransform)
         {
-
-            if (GetLeapController().Devices?.Count >= 1)
+            if (_leapController != null && _leapController.Devices?.Count >= 1)
             {
                 Device currentDevice = _currentDevice;
                 if (currentDevice == null || (_multipleDeviceMode == LeapServiceProvider.MultipleDeviceMode.Specific && currentDevice.SerialNumber != _specificSerialNumber))
                 {
-                    foreach (Device d in GetLeapController().Devices)
```

Comment:
There's a few other places where we call GetLeapController rather than using the private class member - do you think they should also be changed?

Reply:
I found that the other places that GetLeapController is used are at runtime where the deletion of the controller is properly handled so GetLeapController is the most suitable option for those